lambda/translcore.ml

```diff
@@ -471,14 +471,21 @@ and transl_exp0 ~in_new_scope ~scopes e =
                   event_before ~scopes ifso (transl_exp ~scopes ifso),
                   lambda_unit)
   | Texp_sequence(expr1, expr2) ->
-      Lsequence(transl_exp ~scopes expr1,
+      let lam1 = transl_exp ~scopes expr1 in
+      let mode1 = transl_value_mode expr1.exp_mode in
+      let lam1 = maybe_region Alloc_heap [mode1] lam1 in
```

Comment:
I think this should be `[transl_value_mode e.exp_mode]` rather than `Alloc_heap` - we need not introduce a region if the entire `Texp_sequence` is local. (It is of course sound either way)

Reply:
(Just read your comment and saw that this is intentional)